exporter/elasticsearchexporter/integrationtest/exporter_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package integrationtest
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tidwall/gjson"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/plog/plogotlp"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+func TestExporter(t *testing.T) {
+	for _, tc := range []struct {
+		name             string
+		restartCollector bool
+		mockESFailure    bool
+	}{
+		{name: "basic"},
+		{name: "es_intermittent_failure", mockESFailure: true},
+		/* Below tests should be enabled after #30792 is fixed
+		{name: "collector_restarts", restartCollector: true},
+		{name: "collector_restart_with_es_intermittent_failure", mockESFailure: true, restartCollector: true},
+		*/
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			runner(t, tc.restartCollector, tc.mockESFailure)
+		})
+	}
+}
+
+func runner(t *testing.T, restartCollector, mockESFailure bool) {
+	t.Helper()
+
+	cfg := loadConfig(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	flushIvl := time.Second
+	mockES := newMockESClient(t, cfg.Debug)
+	collector := newTestCollector(t, cfg, mockES.ServerURL, flushIvl)
+
+	var g errgroup.Group
+	cancelRun := runTestCollectorWithWait(ctx, t, collector, &g)
+
+	if mockESFailure {
+		mockES.SetReturnStatusCode(http.StatusServiceUnavailable)
+		if !restartCollector {
+			// If restartCollector is not set then recover ES service
+			// after a few failed bulk requests otherwise ensure that
+			// both collector and ES are unavailable for the same
+			// duration considering worst case scenario. We use a
+			// multiple of flush interval to determine the downtime for
+			// ES. Using a multiple greater than 1 ensures that there
+			// is enough downtime to make flush/bulk reqests to ES.
+			time.AfterFunc(10*flushIvl, func() {
+				mockES.SetReturnStatusCode(http.StatusOK)
+			})
+		}
+	}
+
+	count := sendLogs(t, cfg.GRPCEndpoint, "batch_1", 250, 10) // total=logs*agents
+
+	if restartCollector {
+		// Restart the collector after all data is sent to the collector.
+		// Note that the log count should be high to ensure that the collector
+		// stops before all data is shipped to Elasticsearch.
```

Comment:
Is the purpose of this to ensure that the exporter works as expected with the persistent queue? Just asking because this seems to be testing durability, as handled by the persistent queue, more than anything. Would be good to have some more comments about the purpose of this block.

Reply:
> Is the purpose of this to ensure that the exporter works as expected with the persistent queue?

Yeah, mostly it would check durability once we add the code for persistence. I was planning to add another test in the future to test for graceful shutdown (currently `restartExporter` will perform a non-graceful shutdown by canceling the run context immediately). The expectation with graceful shutdown would be to have no data loss given ES is up and running at the time when shutdown happens.

> Would be good to have some more comments about the purpose of this block.

ACK

Reply:
Added a comment explaining `restartExporter` flag.